### PR TITLE
add kbd67mkiirgb v4 support

### PR DESCRIPTION
**Pull request by moyi4681**
add kbd67mkiirgb v4 support

### CODE REVIEW

File: keyboards/kbdfans/kbd67/mkiirgb/v4/config.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#define USB_SUSPEND_WAKEUP_DELAY 5000
+
+#define RGB_DI_PIN C7
+#define RGBLED_NUM 68
+#define DRIVER_LED_TOTAL RGBLED_NUM
+#define RGBLIGHT_ANIMATIONS
+
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/config.h:51]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/kbdfans/kbd67/mkiirgb/v4/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/rules.mk:23]:
```suggestion
RGB_MATRIX_DRIVER = WS2812

```

File: keyboards/kbdfans/kbd67/mkiirgb/v4/v4.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+} };
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void) {
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(30, 0xFF, 0xFF, 0xFF);
+    }
+}
+
+#endif
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/v4.c:49]:
```suggestion
#endif

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd67/mkiirgb/v4/config.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define USB_SUSPEND_WAKEUP_DELAY 5000
+
+#define RGB_DI_PIN C7
+#define RGBLED_NUM 68
+#define DRIVER_LED_TOTAL RGBLED_NUM
+#define RGBLIGHT_ANIMATIONS
+
+#ifdef RGB_MATRIX_ENABLE
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/config.h:49]: Just a heads up, RGB Matrix animation modes are now explicitly enabled, and are not enabled by default. 

```c
#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

File: keyboards/kbdfans/kbd67/mkiirgb/v4/rules.mk [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd67/mkiirgb/v4/config.h [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+#define DEBOUNCE 5
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define USB_SUSPEND_WAKEUP_DELAY 5000
+
+#define RGB_DI_PIN C7
+#define RGBLED_NUM 68
+#define DRIVER_LED_TOTAL RGBLED_NUM
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/config.h:47]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd67/mkiirgb/v4/config.h [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B42
+#define PRODUCT_ID      0x1227
+#define DEVICE_VER      0x0004
+#define MANUFACTURER    KBDFANS
+#define PRODUCT         KBD67MKIIRGB_V4
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd67/mkiirgb/v4/config.h:26]:
```suggestion
#define PRODUCT         KBD67 MKII RGB V4
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by drashna — APPROVED** [review]

**Pull request merged**